# Wine Quality Analysis

This notebook performs an exploratory data analysis (EDA) on the Red Wine Quality dataset.

## 1. Importing Libraries
First, let's import the necessary libraries for data manipulation, analysis, and visualization.

In [ ]:
import numpy as np
import sklearn
import pandas as pd
import matplotlib.pyplot as plt

## 2. Loading the Data
Now, we'll load the dataset from the CSV file into a pandas DataFrame and display the first few rows to get a glimpse of the data.

In [ ]:
df_wine = pd.read_csv('../../Datasets/winequality-red.csv', sep= ',')
df_wine.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

In [ ]:
print(df_wine.shape)

(1599, 12)


## 3. Initial Data Exploration
In this section, we'll perform some initial checks to understand the structure and properties of the dataset.

### 3.1. Dataset Shape
Let's start by checking the dimensions of the dataset (number of rows and columns).

In [ ]:
print(df_wine.shape)

The dataset has 1599 rows and 12 columns. This means there are 11 features (attributes) for each wine sample, plus one target variable ('quality').

### 3.2. Data Types and Non-Null Values
Next, we'll examine the data types of each column and check for any missing values.

In [ ]:
df_wine.info()

The `.info()` output shows that all columns have the `float64` data type, except for the 'quality' column, which is `int64`. This is expected, as quality is a discrete score. Importantly, there are no missing values in any of the columns.

### 3.3. Checking for Missing or Special Values
Although `.info()` reported no nulls, it's good practice to explicitly check for missing values (NA), zeros, or other special values that might indicate data quality issues.

# Questioin 4: howmany missing values do we have?

All of the cells in the df have a value. There are also no values with a value of -1 and there are a few cells with a value of 0 but these make sence for the atribute


In [ ]:
print(df_wine.nunique())

fixed acidity            96
volatile acidity        143
citric acid              80
residual sugar           91
chlorides               153
free sulfur dioxide      60
total sulfur dioxide    144
density                 436
pH                       89
sulphates                96
alcohol                  65
quality                   6
dtype: int64


In [ ]:
df_wine.describe()

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    1599.000000       1599.000000  1599.000000     1599.000000   
mean        8.319637          0.527821     0.270976        2.538806   
std         1.741096          0.179060     0.194801        1.409928   
min         4.600000          0.120000     0.000000        0.900000   
25%         7.100000          0.390000     0.090000        1.900000   
50%         7.900000          0.520000     0.260000        2.200000   
75%         9.200000          0.640000     0.420000        2.600000   
max        15.900000          1.580000     1.000000       15.500000   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  1599.000000          1599.000000           1599.000000  1599.000000   
mean      0.087467            15.874922             46.467792     0.996747   
std       0.047065            10.460157             32.895324     0.001887   
min       0.012000             1.000000              6.000000     0.990070   
25%       0.070000             7.000000             22.000000     0.995600   
50%       0.079000            14.000000             38.000000     0.996750   
75%       0.090000            21.000000             62.000000     0.997835   
max       0.611000            72.000000            289.000000     1.003690   

                pH    sulphates      alcohol      quality  
count  1599.000000  1599.000000  1599.000000  1599.000000  
mean      3.311113     0.658149    10.422983     5.636023  
std       0.154386     0.169507     1.065668     0.807569  
min       2.740000     0.330000     8.400000     3.000000  
25%       3.210000     0.550000     9.500000     5.000000  
50%       3.310000     0.620000    10.200000     6.000000  
75%       3.400000     0.730000    11.100000     6.000000  
max       4.010000     2.000000    14.900000     8.000000